packages/atlas/src/providers/networkUtils/networkUtils.provider.tsx

```diff
@@ -0,0 +1,253 @@
+import { useUser } from '@/providers/user/user.hooks'
+
+const NetworkUtilsContext = createContext<NetworkUtilsContextValue | undefined>(undefined)
+NetworkUtilsContext.displayName = 'NetworkUtilsContext'
+
+export const NetworkUtilsProvider = ({ children }: { children: ReactNode }) => {
+  const client = useApolloClient()
+  const { memberId: activeMemberId } = useUser()
+
+  /*      Channel        */
+
+  const refetchChannel = useCallback(
+    (channelId: string) => {
+      return client.query<GetFullChannelQuery, GetFullChannelQueryVariables>({
+        query: GetFullChannelDocument,
+        variables: {
+          id: channelId,
+        },
+        fetchPolicy: 'network-only',
+      })
+    },
+    [client]
+  )
+
+  /*      Video        */
+
+  const refetchComment = useCallback(
+    (id: string) => {
+      return client.query<GetCommentQuery, GetCommentQueryVariables>({
+        query: GetCommentDocument,
+        variables: {
+          commentId: id,
+        },
+        fetchPolicy: 'network-only',
+      })
+    },
+    [client]
+  )
+
+  const refetchEdits = useCallback(
+    (id: string) => {
+      return client.query<GetCommentEditsQuery, GetCommentEditsQueryVariables>({
+        query: GetCommentEditsDocument,
+        variables: {
+          commentId: id,
+        },
+        fetchPolicy: 'network-only',
+      })
+    },
+    [client]
+  )
+
+  const refetchReactions = useCallback(
+    (videoId: string, memberId?: string) => {
+      return client.query<GetUserCommentsReactionsQuery, GetUserCommentsReactionsQueryVariables>({
+        query: GetUserCommentsReactionsDocument,
+        variables: {
+          memberId: memberId ?? activeMemberId ?? '',
+          videoId: videoId,
+        },
+        fetchPolicy: 'network-only',
+      })
+    },
+    [activeMemberId, client]
+  )
+
+  const refetchReplies = useCallback(
+    (parentCommentId: string) => {
+      return client.query<GetCommentRepliesConnectionQuery, GetCommentRepliesConnectionQueryVariables>({
+        query: GetCommentRepliesConnectionDocument,
+        variables: {
+          parentCommentId,
+        },
+        fetchPolicy: 'network-only',
+      })
+    },
+    [client]
+  )
+
+  const refetchCommentsSection = useCallback(
+    (videoId: string, memberId?: string) => {
+      return client.query<
+        GetUserCommentsAndVideoCommentsConnectionQuery,
+        GetUserCommentsAndVideoCommentsConnectionQueryVariables
+      >({
+        query: GetUserCommentsAndVideoCommentsConnectionDocument,
+        variables: {
+          memberId: memberId ?? activeMemberId,
+          videoId: videoId,
+        },
+        fetchPolicy: 'network-only',
+      })
+    },
+    [activeMemberId, client]
+  )
+
+  const refetchAllCommentsSections = useCallback(async () => {
+    await client.refetchQueries({
+      updateCache: (cache) => {
+        cache.evict({
+          fieldName: 'commentsConnection',
```

Comment:
I'm not sure how to improve this... Ideally fetching the comments by video id would be enough for Apollo to figure out the right query to update, but I don't know if it's doable.

Reply:
I've tried to just refetch comments for specific video ID, but I think because we use `orderBy` and other pagination params in the query, a query with simply video ID doesn't match the same cache entry in Apollo and it doesn't update it.